providers/auth0.go

```diff
@@ -0,0 +1,97 @@
+package providers
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/pusher/oauth2_proxy/pkg/apis/sessions"
+	"github.com/pusher/oauth2_proxy/pkg/requests"
+)
+
+type Auth0Provider struct {
+	*ProviderData
+	Domain string
+}
+
+func NewAuth0Provider(p *ProviderData) *Auth0Provider {
+	p.ProviderName = "Auth0"
+	p.Scope = "openid profile email"
+	return &Auth0Provider{ProviderData: p}
+}
+
+func (p *Auth0Provider) Configure(domain string) {
+	if domain == "" {
+		panic(fmt.Sprintf("auth0 domain not set"))
+	}
+	p.Domain = domain
```

Comment:
Why do you need another param? It simplifies it indeed but it adds another generic option only used for oauth. Looks like simplifying it only makes documentation more complex. Maybe in another PR?